### DISCUSSION TITLE
CSS Battles – Profiles, TIL, Interesting Finds, & Code Golfing Tips

### DISCUSSION THREAD

**Discussion by meg-gutshall** [Virtual-Coffee/virtualcoffee.io#911] [CSS Battles]
During Frontend Friday Folks Fighting CSSBattle.dev, above all we aim to learn about CSS and HTML. We've uncovered some interesting finds over the weeks we've been meeting, and now we have a place to share that knowledge!

Go ahead and add comments here on any TILs (Today I Learned), interesting finds, and code golfing tips you've come across. Drop a link to your [CSSBattle.dev](https://cssbattle.dev/) profile as well. And remember to join us during Frontend Friday Folks fighting CSSBattle.dev every Friday at 11:00 Eastern and Saturday at noon Eastern. Check the Virtual Coffee [events page](https://virtualcoffee.io/events) to see your local time zone.

**Comment by meg-gutshall** [comment]
## Profiles

Share your [CSSBattle.dev](https://cssbattle.dev/) profile in this thread!

To find it, log in to [CSSBattle.dev](https://cssbattle.dev/). Click on the dropdown menu with your username and avatar in the top right of the page. Select **View Profile**. Copy the URL and paste it in a reply on this thread.

**Reply by meg-gutshall** [reply]
https://cssbattle.dev/player/meg_gutshall

**Reply by danieltott** [reply]
https://cssbattle.dev/player/danieltott

**Reply by danieltott** [reply]
If you share here, you can add other people as friends on cssbattle.dev!

**Reply by ClJarvis** [reply]
https://cssbattle.dev/player/jarvisscript

**Reply by radturkin** [reply]
https://cssbattle.dev/player/radturkin

**Reply by david-001** [reply]
https://cssbattle.dev/player/david_001

**Reply by Narigo** [reply]
https://cssbattle.dev/player/narigo

**Reply by michaella23** [reply]
[Micha Rodriguez](https://cssbattle.dev/player/micha_codes)

**Reply by surajb0710** [reply]
https://cssbattle.dev/player/surajb

**Reply by tkshill** [reply]
https://cssbattle.dev/player/specksynder

**Reply by funbunch** [reply]
https://cssbattle.dev/player/funbunch

**Reply by JoeKarow** [reply]
https://cssbattle.dev/player/joekarow

**Reply by missatrox44** [reply]
https://cssbattle.dev/player/missatrox44

**Comment by meg-gutshall** [comment]
## Code Golfing Tips

Share your code golfing tips and tricks in replies on this thread. Remember to use backticks (`` ` ``) to add `inline code` or...

````md
```html
code blocks
```
````

when needed! Also, feel free to include links.

**Reply by meg-gutshall** [reply]
[Tips & Tricks | CSSBattle](https://cssbattle.dev/tips)

**Reply by danieltott** [reply]
[`unit-golf`](https://www.npmjs.com/package/unit-golf)

This is a command-line utility that you can install on your machine that lets you see if any px-based units can be converted into another unit that is smaller! For instance, `100px` can be switched to `25vw`, safely saving 1 character every time you use it!

![Screenshot 2023-06-05 at 1 34 58 PM](https://github.com/Virtual-Coffee/virtualcoffee.io/assets/360261/dc1a6933-e297-4f8a-a20b-ae49c98804cc)

**Reply by danieltott** [reply]
[Minify HTML](https://codebeautify.org/minify-html)

A handy online minifier/beautifier - paste in your normal html and hit minify to get a quick minification of your code, or paste in minified code and hit beautify to look at your minified code in an easier manner.

**Reply by meg-gutshall** [reply]
I found a blog post with some great tips! [Four tips for higher CSS Battle scores. Got any others?](https://www.sitepoint.com/community/t/four-tips-for-higher-css-battle-scores-got-any-others/331431)

**Comment by meg-gutshall** [comment]
# Helpful Resources

These are links we regularly use when solving a CSS Battle. Add more in the thread, and we'll categorize them here in the original post.

## General

- [MDN Web Docs](https://developer.mozilla.org/)
- [CSS-Tricks](https://css-tricks.com/): Tips, Tricks, and Techniques on using Cascading Style Sheets
- [CSS Reference](https://cssreference.io/): A free visual guide to CSS

## Borders & Outline

<details>
<summary><code>border-radius</code></summary><br />

- [MDN – `border-radius`](https://developer.mozilla.org/docs/Web/CSS/border-radius)

</details>

<details>
<summary><code>outline</code></summary><br />

- [MDN – `outline`](https://developer.mozilla.org/docs/Web/CSS/outline)

</details>

## Box Model

<details>
<summary><code>box-sizing</code></summary><br />

- [MDN – `box-sizing`](https://developer.mozilla.org/docs/Web/CSS/box-sizing)

</details>

## Colors

- [Named Colors and Hex Equivalents](https://css-tricks.com/snippets/css/named-colors-and-hex-equivalents/): Sometimes the color name is shorter than the HEX code!

### Gradients

- [MDN – `<gradient>`](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient)
- [MDN – `linear-gradient()`](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient)
- [MDN – `radial-gradient()`](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/radial-gradient)
- [MDN – `repeating-linear-gradient()`](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/repeating-linear-gradient)
- [MDN – `repeating-radial-gradient()`](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/repeating-radial-gradient)
- [MDN – `conic-gradient()`](https://developer.mozilla.org/docs/Web/CSS/gradient/conic-gradient)
- [A Complete Guide to CSS Gradients](https://css-tricks.com/a-complete-guide-to-css-gradients/)
- [A Deep CSS Dive into Radial and Conic Gradients](https://www.smashingmagazine.com/2022/01/css-radial-conic-gradient/)

<details>
<summary><code>mix-blend-mode</code></summary><br />

- [MDN – `mix-blend-mode`](https://developer.mozilla.org/docs/Web/CSS/mix-blend-mode)

</details>

## Display

<details>
<summary><strong><code>flex</code></strong></summary><br />

- [MDN – `flex`](https://developer.mozilla.org/docs/Web/CSS/flex)
- [A Complete Guide to Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)
- [Flexy Boxes](https://the-echoplex.net/flexyboxes/): CSS flexbox playground and code generation tool
- [Flexbox Froggy](https://flexboxfroggy.com/): A game for learning CSS flexbox

</details>

<details>
<summary><strong><code>grid</code></strong></summary><br />

- [MDN – `grid`](https://developer.mozilla.org/docs/Web/CSS/grid)
- [A Complete Guide to CSS Grid](https://css-tricks.com/snippets/css/complete-guide-grid/)
- [Mastering CSS Grid Tutorial](https://www.coltsteele.com/tutorials/mastering-css-grid)
- [Learn CSS Grid for Free](https://scrimba.com/learn/cssgrid)
- [CSS Grid Generator](https://cssgrid-generator.netlify.app/)
- [Layoutit Grid](https://grid.layoutit.com/): CSS Grids layouts made easy!

</details>

<details>
<summary><code>gap</code></summary><br />

- [MDN – `gap`](https://developer.mozilla.org/docs/Web/CSS/gap)

</details>

<details>
<summary><code>place-content</code></summary><br />

- [MDN – `place-content`](https://developer.mozilla.org/docs/Web/CSS/place-content)

</details>

<details>
<summary><code>place-items</code></summary><br />

- [MDN – `place-items`](https://developer.mozilla.org/docs/Web/CSS/place-items)

</details>

## Positioning

<details>
<summary><code>inset</code></summary><br />

- [MDN – `inset`](https://developer.mozilla.org/docs/Web/CSS/inset)

</details>

<details>
<summary><strong><code>position</code></strong></summary><br />

- [MDN – `position`](https://developer.mozilla.org/docs/Web/CSS/position)

</details>

<details>
<summary><code>z-index</code></summary><br />

- [MDN – `z-index`](https://developer.mozilla.org/docs/Web/CSS/z-index)

</details>

## Pseudo-elements

- [`::before` / `::after`](https://css-tricks.com/almanac/selectors/a/after-and-before/)

## Shapes

<details>
<summary><code>clip-path</code></summary><br />

- [MDN – `clip-path`](https://developer.mozilla.org/docs/Web/CSS/clip-path)
- [Clippy](https://bennettfeely.com/clippy/): CSS `clip-path` maker

</details>

### Triangles

- [Drawing a Triangle with CSS](https://alvaromontoro.com/blog/67970/drawing-a-triangle-with-css)

### Half Circles

- [CSS Half Circle](https://unused-css.com/blog/css-half-circle/)

## Transforms

<details>
<summary><code>rotate</code></summary><br />

- [MDN – `rotate`](https://developer.mozilla.org/docs/Web/CSS/rotate)

</details>

**Reply by piotrszymaniec** [reply]
great summary! Thanks Meg

**Reply by danieltott** [reply]
MDN is far and away the best place to dive into these sorts of things!

**Comment by ClJarvis** [comment]
I wrote a summary of CSS Battles. If you're new to the group read it for an overview.
[CSS Battle Test your Skills](
https://dev.to/jarvisscript/css-battle-test-your-skills-1ia6)

**Reply by meg-gutshall** [reply]
I just saw this Chris, way cool! We'll have to use this to promote our weekly battles!
